ext/src/ruby_api/wasi_config.rs

```diff
@@ -233,6 +293,39 @@ impl WasiConfig {
         rb_self
     }
 
+    /// @yard
+    /// Set mapped directory for host path and guest path.
+    /// @param host_path [String]
+    /// @param guest_path [String]
+    /// @param dir_perms [Symbol] Directory permissions, one of :read, :mutate, or :all
+    /// @param file_perms [Symbol] File permissions, one of :read, :write, or :all
+    /// @def set_mapped_directory(host_path, guest_path, dir_perms, file_perms)
+    /// @return [WasiConfig] +self+
+    pub fn set_mapped_directory(
+        rb_self: RbSelf,
+        host_path: RString,
+        guest_path: RString,
+        dir_perms: Symbol,
+        file_perms: Symbol,
+    ) -> Result<RbSelf, Error> {
+        let host_path = host_path.to_string().unwrap();
+        let guest_path = guest_path.to_string().unwrap();
+        let dir_perms: DirPerms = PermsSymbolEnum(dir_perms).try_into()?;
+        let file_perms: FilePerms = PermsSymbolEnum(file_perms).try_into()?;
+
+        let mapped_dir = MappedDirectory {
+            host_path,
+            guest_path,
+            dir_perms,
+            file_perms,
+        };
+
+        let mut inner = rb_self.inner.borrow_mut();
+        inner.mapped_directories.push(mapped_dir);
```

Comment:
To:
* Keep things consistent with the other build options (see e.g., `env`, could we make `MappedDirectory` hold the Ruby values rather than eagerly converting them to Rust on each call?
* Related to point 1, that approach will also result in less overhead in case any of the other values fails to validate and centralizing everything in the `build` call.